src/libs/ReportListUtils.js

```diff
@@ -30,63 +47,149 @@ function getParticipantLogins(report) {
+        searchTerms.push(...report.participants);
+    }
+
+    return _.unique(searchTerms).join(' ');
 }
 
 /**
  * Creates a report list option
  *
  * @param {Array<Object>} personalDetailList
  * @param {Object} [report]
- * @param {Object} [comments]
- * @param {Number} [activeReportID]
  * @returns {Object}
  */
-function createOption(personalDetailList, report, comments, activeReportID) {
-    // There is only one participant so return the single participant view
+function createOption(personalDetailList, report) {
+    const hasMultipleParticipants = personalDetailList.length > 1;
     const personalDetail = personalDetailList[0];
-    const isSingleUserDM = personalDetailList.length < 2;
     return {
         text: report ? report.reportName : personalDetail.displayName,
-        alternateText: (!report || isSingleUserDM) ? personalDetail.login : report.reportName,
+        alternateText: (!report || !hasMultipleParticipants) ? personalDetail.login : report.reportName,
         icons: report ? report.icons : [personalDetail.avatarURL],
-        login: personalDetailList.length < 2 ? personalDetail.login : null,
+
+        // It doesn't make sense to provide a login in the case of a report with multiple participants since
+        // there isn't any one single login to refer to for a report.
+        login: !hasMultipleParticipants ? personalDetail.login : null,
         reportID: report ? report.reportID : null,
         isUnread: report ? report.unreadActionCount > 0 : null,
-        hasDraftComment: report && report.reportID !== activeReportID && hasComment(comments, report.reportID),
+        hasDraftComment: report && report.reportID !== activeReportID && hasComment(report.reportID),
         keyForList: report ? String(report.reportID) : personalDetail.login,
+        searchText: getSearchText(report, personalDetailList),
+        isPinned: report && report.isPinned,
     };
 }
 
+/**
+ * Rebuild the options. We are throttling this so the options are only rebuilt at most
+ * once per second. It can be expensive to rebuild options so we slow this down a bit.
+ */
+const rebuildOptions = _.throttle(() => {
+    const reportMapForLogins = {};
+    const orderedReports = lodashOrderBy(reports, ['lastVisitedTimestamp'], ['desc']);
+    allReportOptions = _.map(orderedReports, (report) => {
+        const logins = getParticipantLogins(report);
+        const hasMultipleParticipants = logins.length > 1;
+        const reportPersonalDetails = getPersonalDetailsForLogins(logins);
+        if (!hasMultipleParticipants) {
+            reportMapForLogins[logins[0]] = report;
+        }
+        return createOption(reportPersonalDetails, report);
+    });
+    allPersonalDetailsOptions = _.map(personalDetails, personalDetail => (
+        createOption([personalDetail], reportMapForLogins[personalDetail.login])
+    ));
+}, 1000, {leading: false});
+
+Onyx.connect({
```

Comment:
shouldn't all these onyx connects be at the top?

Reply:
Maybe? Can you give me a hint about why you think they should be?

Reply:
p.s. they are here because the linter wanted me to move the variables they set before the functions that call them. But I didn't think about if the subscribers should be moved.

Reply:
Well in all the libs file i see it placed on the top so i thought that was a format we were following. Having this be different is odd.

Reply:
Discussed 1:1 and its fine to have it here atm.